packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,153 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+
+  /**
+   * Gateway route configuration for rewriting the prefix of the URL path of the request.
+   *
+   * @default - rewrites the request's URL path to '/'
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route matching based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixPathMatch the value to use to match the beginning of the path part of the URL of the request.
+   *   It must start with the '/' character.
+   *   When `rewriteTo` is provided, it must also end with the '/' character.
+   *   If provided as "/", matches all requests.
+   *   For example, if your virtual service name is "my-service.local"
+   *   and you want the route to match requests to "my-service.local/metrics", your prefix should be "/metrics".
+   * @param rewriteTo Specify either disabling automatic rewrite or rewriting to specified prefix path.
+   *   To disable automatic rewrite, provide `''`.
+   *   As a default, request's URL path is automatically rewritten to '/'.
+   */
+  public static startsWith(prefixPathMatch: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, rewriteTo);
+  }
+
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   *   As a default, retains original request's URL path.
+   */
+  public static exactly(path: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, rewriteTo);
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   *   As a default, retains original request's URL path.
+   */
+  public static regex(regex: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex }, rewriteTo);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly rewriteTo?: string,
+  ) {
+    super();
+
+    if (prefixPathMatch[0] !== '/') {
+      throw new Error('Prefix path for the match must start with \'/\', '
+        + `got: ${prefixPathMatch}`);
+    }
+
+    if (rewriteTo && rewriteTo !== '') {
```

Comment:
The second part of the condition is redundant (`''` is falsy in JavaScript):

```suggestion
    if (rewriteTo) {
```

Reply:
That's right. Thanks!